App/src/main/java/com/gluonhq/chat/service/UpdateService.java

```diff
@@ -0,0 +1,132 @@
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class UpdateService {
+
+    private static final String TEMP_DOWNLOAD_DIRECTORY = System.getProperty("java.io.tmpdir");
+    static Path installerPath;
+
+    static String currentAppVersion() {
+        String currentVersion = "1.0.0";
+        Properties about = new Properties();
+        try {
+            about.load(UpdateService.class.getResourceAsStream("/about.properties"));
+            String version = about.getProperty("app.version");
+            if (version != null && !version.isEmpty()) {
+                currentVersion = version;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return currentVersion;
+    }
+
+        static GluonObservableList<GithubRelease> queryReleases() {
+        final RestClient restClient = RestClient.create()
+                .method("GET")
+                .host("https://api.github.com/repos/gluonhq/wave-app/releases");
+        return DataProvider.retrieveList(restClient.createListDataReader(GithubRelease.class));
+    }
+
+    static void downloadNewVersion(GithubRelease githubRelease) {
+        GithubRelease.Asset windowsAsset = githubRelease.getAssets().stream()
+                .filter(asset -> asset.getName().endsWith(OSFileExtension()))
+                .findFirst().get();
+        String download_url = windowsAsset.getBrowser_download_url();
+        Path tempFile = Paths.get(TEMP_DOWNLOAD_DIRECTORY + "/" + download_url.substring(download_url.lastIndexOf("/") + 1));
+        try {
+            if (Files.exists(tempFile) && Files.size(tempFile) == windowsAsset.getSize()) {
+                System.out.println("File already found:"  + tempFile);
+                installerPath = tempFile;
+                return;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        HttpClient client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.ALWAYS).build();
+        URI uri = URI.create(download_url);
+        HttpRequest request = HttpRequest.newBuilder().uri(uri).build();
+
+        // use the client to send the asynchronous request
+        InputStream is = client.sendAsync(request, HttpResponse.BodyHandlers.ofInputStream()).thenApply(HttpResponse::body).join();
+        try (FileOutputStream out = new FileOutputStream(tempFile.toString())) {
+            is.transferTo(out);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println("New Version downloaded at:"  + tempFile);
+        installerPath = tempFile;
+    }
+
+    static void installNewVersion() {
+        try {
+            Runtime.getRuntime().exec("cmd /c start " + installerPath);
```

Comment:
This is windows specific, but we don't test on Windows?

Reply:
Good catch. Instead of an if..else.. block, a better alternative would be to use AWT's ` Desktop.getDesktop().open(installerPath.toFile())`. Thoughts?

Reply:
This will pull in the whole AWT and blow up the appsize when we use native-image
This actually sounds something for Attach. 
But for now, the easiest and least intrusive approach is probably using if/else blocks